packages/postgres-database/src/simcore_postgres_database/utils_repos.py

```diff
@@ -11,6 +11,9 @@
 async def pass_or_acquire_connection(
     engine: AsyncEngine, connection: AsyncConnection | None = None
 ) -> AsyncIterator[AsyncConnection]:
+    """
+    When to use: For READ operations only!
+    """
     # NOTE: When connection is passed, the engine is actually not needed
```

Comment:
can you explain why? and also the one below?

Reply:
I will leave this one for @pcrespov to answer, please.

Reply:
Well, this seems to me more of a "short practical usage note" to distinguish when to use `pass_or_acquire_connection` and when `transaction_context` . 


Here a better doc for `pass_or_acuiqre_connection`

```
It ensures that a connection is available for use within the context, either by using an existing connection passed as a parameter or by acquiring a new one from the engine. The caller must manage the lifecycle of any connection explicitly passed in, but the function handles the cleanup for connections it creates itself. This function **does not open new transactions** and therefore is recommended only for read-only database operations. 
```

For `transaction_context` 

```
This function manages the database connection and ensures that a transaction context is established for write operations. It supports both outer and nested transactions, providing flexibility for scenarios where transactions may already exist in the calling context.
```